x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/entity_descriptions/generic.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { newestValue } from './field_utils';
+import type { EntityDescription } from '../types';
+import { getCommonFieldDescriptions } from './common';
+
+export const GENERIC_DEFINITION_VERSION = '1.0.0';
+export const GENERIC_IDENTITY_FIELD = 'entity.id';
+export const genericEntityEngineDescription: EntityDescription = {
+  entityType: 'generic',
+  version: GENERIC_DEFINITION_VERSION,
+  identityField: GENERIC_IDENTITY_FIELD,
+  settings: {
+    timestampField: '@timestamp',
+  },
+  fields: [
+    newestValue({ source: 'entity.name' }),
+    newestValue({ source: 'entity.source' }),
+    newestValue({ source: 'entity.category' }),
+    newestValue({ source: 'entity.type' }),
+    newestValue({ source: 'entity.address' }),
+
+    newestValue({ source: 'cloud.account.id' }),
+    newestValue({ source: 'cloud.account.name' }),
+    newestValue({ source: 'cloud.availability_zone' }),
+    newestValue({ source: 'cloud.instance.id' }),
+    newestValue({ source: 'cloud.instance.name' }),
+    newestValue({ source: 'cloud.machine.type' }),
+    newestValue({ source: 'cloud.project.id' }),
+    newestValue({ source: 'cloud.project.name' }),
+    newestValue({ source: 'cloud.provider' }),
+    newestValue({ source: 'cloud.region' }),
+    newestValue({ source: 'cloud.service.name' }),
+
+    newestValue({ source: 'host.architecture' }),
+    newestValue({ source: 'host.boot.id' }),
+    newestValue({ source: 'host.cpu.usage' }),
+    newestValue({ source: 'host.disk.read.bytes' }),
+    newestValue({ source: 'host.disk.write.bytes' }),
```

Comment:
I'm wondering how reconciliation works in this context, for ex how `newestValue` will work on the Generic entity, this means we will have the newest known value according to the `entity.id` so only integrations that sends the `entity.id` alongside the data will be able to keep these values up to date? For example, if host data can be spread among other logs this means only the "Host" Entity Store will have the most up-to-date field for a host as it uses the host.id for reconciliation, while the host information in the Generic entity store will have to wait for an integration that sends the `entity.id` + `entity.type` information alongside the host

Reply:
You are correct on your understanding.